pennylane_lightning/core/_serialize.py

```diff
@@ -269,22 +267,22 @@ def _pauli_sentence(self, observable, wires_map: dict = None):
     # pylint: disable=protected-access, too-many-return-statements
     def _ob(self, observable, wires_map: dict = None):
         """Serialize a :class:`pennylane.operation.Observable` into an Observable."""
-        if isinstance(observable, (Prod, Sum, SProd)) and observable.pauli_rep is not None:
+        if isinstance(observable, (PauliX, PauliY, PauliZ, Identity, Hadamard)):
+            return self._named_obs(observable, wires_map)
+        if isinstance(observable, Hamiltonian):
+            return self._hamiltonian(observable, wires_map)
+        if observable.pauli_rep is not None:
             return self._pauli_sentence(observable.pauli_rep, wires_map)
         if isinstance(observable, Tensor) or (
             isinstance(observable, Prod) and not observable.has_overlapping_wires
         ):
             return self._tensor_ob(observable, wires_map)
-        if observable.name in ("Hamiltonian", "LinearCombination"):
+        if isinstance(observable, (LinearCombination, Prod, SProd, Sum)):
+            # if isinstance(observable, (Sum, Prod, SProd)):
+            #     observable = observable.simplify()
             return self._hamiltonian(observable, wires_map)
         if observable.name == "SparseHamiltonian":
```

Comment:
is there any issue with checking the type instead of obs name string repr here?
```suggestion
        if isinstance(observable, qml.SparseHamiltonian):
```